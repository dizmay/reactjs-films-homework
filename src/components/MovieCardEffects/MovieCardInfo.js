import React from 'react';
import PropTypes from 'prop-types';
import ButtonElement from '../ButtonElement/ButtonElement';
import MovieInfo from '../MovieInfo/MovieInfo';
import styles from './MovieCardInfo.scss';

const MovieCardInfo = ({
  info, showInfo, movieInfo, showTrailer,
}) => (
  <div className={info ? styles.info__container : styles.info__container_hide}>
    <button type="button" onClick={showInfo} className={styles.info__close}>x</button>
    <MovieInfo
      title={movieInfo.title}
      rate={movieInfo.rate}
      genres={movieInfo.genres}
      white
    />
    <span className={styles.info_overview}>{movieInfo.overview}</span>
    <ButtonElement blue clickFunc={showTrailer}>Watch Now</ButtonElement>
  </div>
);

MovieCardInfo.defaultProps = {
  info: false,
  showInfo: null,
  movieInfo: {},
  showTrailer: null,
};

MovieCardInfo.propTypes = {
  info: PropTypes.bool,
  showInfo: PropTypes.func,
  movieInfo: PropTypes.objectOf(PropTypes.any),
  showTrailer: PropTypes.func,
};

export default MovieCardInfo;
