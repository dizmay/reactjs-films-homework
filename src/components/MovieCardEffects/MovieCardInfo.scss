* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.info__container {
  position: absolute;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  background: #000000b3;
  padding: 1rem 1rem 4rem;
}

.info__container_hide {
  display: none;
}

.info__close {
  width: 2rem;
  height: 2rem;
  color: #fff;
  font-size: 2rem;
  background: transparent;
  border: none;
  outline: none;
  cursor: pointer;
}

.info_overview {
  height: 7.5rem;
  color: #fff;
  overflow: auto;
  text-overflow: ellipsis;
  margin-bottom: 0.5rem;
}
