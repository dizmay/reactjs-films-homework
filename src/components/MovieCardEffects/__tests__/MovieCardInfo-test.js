import React from 'react';
import renderer from 'react-test-renderer';
import MovieCardInfo from '../MovieCardInfo';

it('renders correctly', () => {
  const tree = renderer.create(<MovieCardInfo />).toJSON();
  expect(tree).toMatchSnapshot();
});
