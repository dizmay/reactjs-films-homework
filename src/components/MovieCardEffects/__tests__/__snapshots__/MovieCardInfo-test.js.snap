// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="info__container_hide"
>
  <button
    className="info__close"
    onClick={null}
    type="button"
  >
    x
  </button>
  <div
    className="info__container"
  >
    <div
      className="info"
    >
      <span
        className="info__title info__title_black info__title_white"
      >
        
      </span>
      <div
        className="info__rate"
      >
        0.0
      </div>
    </div>
    <div
      className="info__genres"
    >
      
    </div>
  </div>
  <span
    className="info_overview"
  />
  <button
    className="btn blue"
    onClick={null}
    type="button"
  >
    Watch Now
  </button>
</div>
`;
